services/dataplatform/nodepool.go

```diff
@@ -0,0 +1,317 @@
+	"log"
+	"strings"
+)
+
+var nodePoolResourceName = "Dataplatform Node Pool"
+
+func (c *Client) GetNodePool(ctx context.Context, clusterId, nodePoolId string) (dataplatform.NodePoolResponseData, *dataplatform.APIResponse, error) {
+	cluster, apiResponse, err := c.sdkClient.DataPlatformNodePoolApi.GetClusterNodepool(ctx, clusterId, nodePoolId).Execute()
+	apiResponse.LogInfo()
+	return cluster, apiResponse, err
+}
+
+func (c *Client) IsNodePoolDeleted(ctx context.Context, d *schema.ResourceData) (bool, error) {
+	clusterId, ok := d.GetOk("cluster_id")
+	if !ok {
+		return false, fmt.Errorf("could not get cluster_id from schema")
+	}
+	clusterIdStr := clusterId.(string)
+	nodePoolId := d.Id()
+	_, apiResponse, err := c.sdkClient.DataPlatformNodePoolApi.GetClusterNodepool(ctx, clusterIdStr, nodePoolId).Execute()
+	apiResponse.LogInfo()
+	return apiResponse.HttpNotFound(), err
+}
+
+func (c *Client) ListNodePools(ctx context.Context, clusterId string) (dataplatform.NodePoolListResponseData, *dataplatform.APIResponse, error) {
+	nodePool, apiResponse, err := c.sdkClient.DataPlatformNodePoolApi.GetClusterNodepools(ctx, clusterId).Execute()
+	apiResponse.LogInfo()
+	return nodePool, apiResponse, err
+}
+
+func (c *Client) CreateNodePool(ctx context.Context, clusterId string, d *schema.ResourceData) (dataplatform.NodePoolResponseData, *dataplatform.APIResponse, error) {
+	dataplatformNodePool := GetDataplatformNodePoolDataCreate(d)
+	clusterResponse, apiResponse, err := c.sdkClient.DataPlatformNodePoolApi.CreateClusterNodepool(ctx, clusterId).CreateNodePoolRequest(*dataplatformNodePool).Execute()
+	apiResponse.LogInfo()
+	return clusterResponse, apiResponse, err
+}
+
+func (c *Client) UpdateNodePool(ctx context.Context, clusterId, nodePoolId string, d *schema.ResourceData) (dataplatform.NodePoolResponseData, utils.ApiResponseInfo, error) {
+	dataplatformNodePool := GetDataplatformNodePoolDataUpdate(d)
+	clusterResponse, apiResponse, err := c.sdkClient.DataPlatformNodePoolApi.PatchClusterNodepool(ctx, clusterId, nodePoolId).PatchNodePoolRequest(*dataplatformNodePool).Execute()
+	apiResponse.LogInfo()
+	return clusterResponse, apiResponse, err
+}
+
+func (c *Client) DeleteNodePool(ctx context.Context, clusterId, nodePoolId string) (dataplatform.NodePoolResponseData, *dataplatform.APIResponse, error) {
+	clusterResponse, apiResponse, err := c.sdkClient.DataPlatformNodePoolApi.DeleteClusterNodepool(ctx, clusterId, nodePoolId).Execute()
+	apiResponse.LogInfo()
+	return clusterResponse, apiResponse, err
+}
+func (c *Client) IsNodePoolReady(ctx context.Context, d *schema.ResourceData) (bool, error) {
+	clusterId, ok := d.GetOk("cluster_id")
+	if !ok {
+		return false, fmt.Errorf("could not get cluster_id from schema")
+	}
+	clusterIdStr := clusterId.(string)
+	subjectNodePool, _, err := c.GetNodePool(ctx, clusterIdStr, d.Id())
+	if err != nil {
+		return true, fmt.Errorf("checking Dataplatform Node Pool status: %w", err)
```

Comment:
@cristiGuranIonos `error checking ..`. Also, shouldn't we return `false` here since the node pool is not ready?